### DISCUSSION TITLE
Configure terminus

### DISCUSSION THREAD

**Discussion by onordgren** [directus/directus#17237] [Feature Requests]
Hello! We're running directus in k8s and would like to be able to control the timeout period in the terminus handler. From the source code it looks like it's hard coded at 1000ms. Is there some other recommended way of doing this, was thinking of maybe doing it in the `server.stop` hook. Or would you consider adding a configuration option to be able to set the terminus timeout?

**Comment by rijkvanzanten** [comment]
@onordgren I think we can/should make that configurable from an environment variable. Any ideas / recommendations on a good name for that? I'd rather not tie it to `terminus` directly, as we might swap that out over time (eg keep the env var name a little more generic). Maybe `SERVER_SHUTDOWN_TIMEOUT`? 

For those reading along, the values are here:

https://github.com/directus/directus/blob/276cc479ceb43cec3a374f369afa5842cd5a206b/api/src/server.ts#L84-L90

**Reply by onordgren** [reply]
`SERVER_SHUTDOWN_TIMEOUT` sounds great to me! Follow up question, is the health check end point exposed somewhere? Think it's `/healthcheck` by default from terminus. It usually gives a `200` when ready but then a `503` during shut down.

**Reply by rijkvanzanten** [reply]
Directus has a health endpoint: https://docs.directus.io/reference/system/server.html#health 👍🏻

**Reply by onordgren** [reply]
Ah yea, we are currently using that one but from what I can see it will not return `503` during shutdown. Might be missing something though.

**Reply by onordgren** [reply]
Maybe there should be a `testShutdown` or similar here?
```
checks: merge(
  ...(await Promise.all([testDatabase(), testCache(), testRateLimiter(), testStorage(), testEmail()]))
),
```

**Reply by rijkvanzanten** [reply]
I'm not quite sure what the question is.. You're saying that `/health` should return 503 during shutdowns? I don't understand what you mean with the `testShutdown` example

**Reply by onordgren** [reply]
Terminus exposes a /healthcheck end point by default, see below copy paste from their read me
```
  healthChecks: {
    '/healthcheck': healthCheck,    // a function accepting a state and returning a promise indicating service health,
    ...
  },
  ...
  statusOk,                         // [optional = 200] status to be returned for successful healthchecks
  statusError,                      // [optional = 503] status to be returned for unsuccessful healthchecks
```
This endpoint will return `200` by default but once the shutdown has been triggered it will return `503` during the timeout. During the timeout the app itself can still handle regular traffic but this end point is used to tell k8s that the app is no longer healthy.

I don't think that the directus `/health` end point takes this into account so it cannot be used as a way to determine if the app is shutting down or not. One possible solution would be to add a handler like `testShutdown` to it that checks if the app is shutting down or not, kind of what terminus does on `/healthcheck`.

**Reply by rijkvanzanten** [reply]
Ah gotcha! Yeah that's a good thing to resolve on the `/health` endpoint. I'd rather not add a secondary `/healthcheck` endpoint. Created an issue to track here: https://github.com/directus/directus/issues/17241

**Comment by rijkvanzanten** [comment]
This feature request has been accepted and is queued to be implemented! You can follow along with the progress here: #17239

**Reply by onordgren** [reply]
Awesome! Thx for the swift feedback.

**Reply by rijkvanzanten** [reply]
> Awesome! Thx for the swift feedback.

Super easy tweak, big QOL improvement for k8s use 🙂
